blockstore.go

```diff
@@ -107,32 +107,56 @@ type gcBlockstore struct {
 	GCLocker
 }
 
-// NewBlockstore returns a default Blockstore implementation
-// using the provided datastore.Batching backend.
-func NewBlockstore(d ds.Batching) Blockstore {
-	var dsb ds.Batching
-	dd := dsns.Wrap(d, BlockPrefix)
-	dsb = dd
-	return &blockstore{
-		datastore: dsb,
-		rehash:    uatomic.NewBool(false),
+// Option is a default implementation Blockstore option
+type Option struct {
+	f func(bs *blockstore)
+}
+
+// WriteThrough skips checking if the blockstore already has a block before
+// writing it.
+func WriteThrough() Option {
+	return Option{
+		func(bs *blockstore) {
+			bs.writeThrough = true
+		},
 	}
 }
 
-// NewBlockstoreNoPrefix returns a default Blockstore implementation
+// NoPrefix avoids wrapping the blockstore into the BlockPrefix namespace
+// ("/blocks"), so keys will not be modified in any way.
+func NoPrefix() Option {
+	return Option{
+		func(bs *blockstore) {
+			bs.noPrefix = true
+		},
+	}
+}
+
+// NewBlockstore returns a default Blockstore implementation
 // using the provided datastore.Batching backend.
-// This constructor does not modify input keys in any way
-func NewBlockstoreNoPrefix(d ds.Batching) Blockstore {
```

Comment:
I will push a stub `NewBlockstoreNoPrefix` that call `NewBlockstore(d, NoPrefix())` under the hood.
No need to have an API breaking change here (I know the variadic technically break the API but that only if you use reflection).